lib/adapters/adapter.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Janus.Transport.WS.Adapter do
+  @moduledoc """
+  This module takes part in communicating `Janus.Transport.WS` module
+  with lower level WebSocket client (e.g. `:websockex`).
+
+  It is responsible for sending and passing back messages, notifying about socket status change.
+
+  Sending messages is supposed to be synchronous while receiving is asynchronous.
```

Comment:
Otherwise WS transport will never return error on message send when called from `Connection` module. Therefore we will need to handle asynchronous error in some other place where it will not be so obvious where it came from and from which call. If those are not problems then it might as well be async.

Reply:
Changed it to be async.